### DISCUSSION TITLE
Downloading All My Gists From Github At Once

### DISCUSSION THREAD

**Discussion by ghost** [community/community#23895] [New to GitHub]
<p>Greetings Everyone out there! I’m new to Github. I have 41 Gists in my account and I want to have a copy of all those gists in my local disk. I searched everywhere but could not find any solution to my question. There are ways to download gists one by one but downloading all of it 1 by 1 is difficult &amp; impractical. Could you please help me by giving a solution? :pray:</p>
<p>Thanks,<br>
MrStarkJr</p>

**Comment by jfullstackdev** [comment]
<aside class="onebox allowlistedgeneric" data-onebox-src="https://github.com/cuducos/getgist">
  <header class="source">
      <img src="https://github.githubassets.com/favicons/favicon.svg" class="site-icon" width="32" height="32">

      <a href="https://github.com/cuducos/getgist" target="_blank" rel="noopener nofollow ugc">GitHub</a>
  </header>

  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/345;"><img src="https://user-images.githubusercontent.com/47092464/181079875-b9003097-ca50-48b3-b4fc-b649c6e6adff.png" class="thumbnail" width="690" height="345" srcset="https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d096171700303394b94ac60a13d3a91b80f69951_2_690x345.png, https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d096171700303394b94ac60a13d3a91b80f69951_2_1035x517.png 1.5x, https://aws1.discourse-cdn.com/github/original/3X/d/0/d096171700303394b94ac60a13d3a91b80f69951.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/3X/d/0/d096171700303394b94ac60a13d3a91b80f69951_2_10x10.png"></div>

<h3><a href="https://github.com/cuducos/getgist" target="_blank" rel="noopener nofollow ugc">GitHub - cuducos/getgist: 🖥️ Easily download any file from a GitHub Gist,...</a></h3>

  <p>🖥️ Easily download any file from a GitHub Gist, with one single command. - GitHub - cuducos/getgist: 🖥️ Easily download any file from a GitHub Gist, with one single command.</p>


  </article>

  <div class="onebox-metadata">
    
    
  </div>

  <div style="clear: both"></div>
</aside>

<p>maybe this is the app that you are looking for,</p>

**Comment by airtower-luna** [comment, marked as answer]
<p>If you’d rather script something yourself, here’s the REST API documentation for Gist access: <a href="https://docs.github.com/en/rest/reference/gists" class="inline-onebox">Gists - GitHub Docs</a></p>
<p>You can list your Gists and then either download the files, or clone the repositories. :slightly_smiling_face:</p>

**Comment by irving-caamal** [comment]
<p>Nice solution, are you using <a href="https://docs.github.com/en/rest/reference/gists" rel="noopener nofollow ugc">Gists API?</a> ?</p>

**Comment by devarda** [comment]
I came here, couldn't find readily available code that used GitHub API and I wrote this: https://github.com/devarda/github-takeout 

You may download all your Gists that you've created, all the Gists you have forked, all the Gists you have starred. You can list them as CSV to have an index or just to sort through. It will download all the Gists into separate folders and will place each file inside. The scripts are simple and hackable so that you can add more meta in CSV or create a index.md file for each Gist to store meta - up to you to adjust to your needs!

Here are some examples of what you can do:

```bash
# download forked gists
yarn download-forked-gists

# download your gists
yarn download-my-gists

# download starred gists
yarn download-starred-gists

# list gists with files
yarn list-gists-with-files

# list gists
yarn list-gists

# etc.
```

**Comment by cgoldberg** [comment]
https://github.com/cgoldberg/githubtakeout

**Comment by ShradhaKhard** [comment]
- To list your gists
```
gh api \
  --method GET \
  -H "Accept: application/vnd.github+json" \
  -H "X-GitHub-Api-Version: 2026-03-10" \
  -F per_page=100 \
  /gists
```

- To download all the gists as files
```
mkdir -p gists && cd gists
gh api --paginate "/gists?per_page=100" \
  --jq '.[] | .id as $id | .files[] | [$id, .filename, .raw_url] | @tsv' | \
while IFS=$'\t' read -r id name url; do
  curl -sL "$url" -o "${id}_${name}"
done
```

- To check how many you downloaded 
```
gh api --paginate "/gists?per_page=100" --jq '.[].id' | wc -l
```

- To bulk delete the gists
```
gh api --paginate "/gists?per_page=100" --jq '.[].id' | \
  xargs -I{} gh api -X DELETE "/gists/{}"
```
